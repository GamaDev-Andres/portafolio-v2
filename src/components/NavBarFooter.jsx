import ListOfLinks from './ListOfLinks'

const NavBarFooter = () => {
  return (
    <nav
      id="navFooter"
      className="flex w-full md:hidden fixed bottom-0 transition-transform duration-500 px-4 py-2 border-t z-10 bg-base border-bordes"
    >
      <ListOfLinks ViewInPhone={true} />
    </nav>
  )
}

export default NavBarFooter
