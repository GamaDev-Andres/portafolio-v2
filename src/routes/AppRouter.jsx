import { useState } from 'react';
import { BrowserRouter, Route, Routes } from 'react-router-dom';
import Home from '../components/home/Home';
import Project from '../components/project/Project';

function AppRouter() {
  return (
    <BrowserRouter>
      <Routes>
        <Route path="/" element={<Home />} />
        <Route path="/project/:id" element={<Project />} />
      </Routes>
    </BrowserRouter>
  );
}

export default AppRouter;
