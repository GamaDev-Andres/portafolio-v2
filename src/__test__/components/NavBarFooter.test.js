/**
 * @jest-environment jsdom
 */
import Enzyme from 'enzyme'
import { HelmetProvider } from 'react-helmet-async'
import { MemoryRouter } from 'react-router-dom'
import NavBarFooter from '../../components/NavBarFooter'
import AppRouter from '../../routes/AppRouter'

describe('probando <NavBarFooter/>', () => {
  test('debe montarse correctamente', () => {
    const wrapper = Enzyme.mount(
      <NavBarFooter />
    )
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('nav').find('ListOfLinks').length).toBe(1)
  })

  test('No debe de montarse', () => {
    const wrapper = Enzyme.mount(
      <MemoryRouter initialEntries={['/project/BARBERAPP']}>
        <HelmetProvider>
          <AppRouter />
        </HelmetProvider >
      </MemoryRouter>
    )
    expect(wrapper.find('NavBarFooter').exists()).toBe(false)
  })
})
