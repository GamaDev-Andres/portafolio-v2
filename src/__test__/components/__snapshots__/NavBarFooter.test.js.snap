// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`probando <NavBarFooter/> debe montarse correctamente 1`] = `
<nav
  className="bg-base dark:bg-baseDark flex w-full sm:hidden fixed bottom-0 transition-transform duration-500 px-4 py-2 border-t z-10  border-bordes"
  id="navFooter"
>
  <div
    className="sm:gap-4 flex-grow sm:flex-grow-0 sm:hidden flex w-full justify-between"
  >
    <a
      className="sm:hover:bg-blue-200 dark:sm:hover:bg-gris flex flex-col sm:flex-row items-center transition-colors duration-150 rounded-lg sm:px-4 sm:py-2 py-0 font-medium"
      href="/#inicio"
    >
      <i
        className="sm:mr-2 fa-solid fa-house"
      />
      <span
        className="text-xs"
      >
        inicio
      </span>
    </a>
    <a
      className="sm:hover:bg-blue-200 dark:sm:hover:bg-gris flex flex-col sm:flex-row items-center transition-colors duration-150 rounded-lg sm:px-4 sm:py-2 py-0 font-medium"
      href="/#sobremi"
    >
      <i
        className="sm:mr-2 fa-solid fa-user"
      />
      <span
        className="text-xs"
      >
        sobre m√≠
      </span>
    </a>
    <a
      className="sm:hover:bg-blue-200 dark:sm:hover:bg-gris flex flex-col sm:flex-row items-center transition-colors duration-150 rounded-lg sm:px-4 sm:py-2 py-0 font-medium"
      href="/#habilidades"
    >
      <i
        className="sm:mr-2 fa-solid fa-laptop-code"
      />
      <span
        className="text-xs"
      >
        habilidades
      </span>
    </a>
    <a
      className="sm:hover:bg-blue-200 dark:sm:hover:bg-gris flex flex-col sm:flex-row items-center transition-colors duration-150 rounded-lg sm:px-4 sm:py-2 py-0 font-medium"
      href="/#proyectos"
    >
      <i
        className="sm:mr-2 fa-solid fa-briefcase"
      />
      <span
        className="text-xs"
      >
        proyectos
      </span>
    </a>
  </div>
</nav>
`;
